### PR TITLE
Fix the issue of patch repo in upgrade cmd

### PR DESCRIPTION
**Pull request by d8660091**
*Issue #, if available:*

*Description of changes:* codebuild  aws-eks-anywhere-build-tooling-projects-upgrade is running without RELEASE_BRANCH environment variable, and it cannot run the patch-repo make target. This PR is trying to address the issue.

*Test*: test the code change with BASE_REPO_OWNER=d8660091, and `make upgrade -C tools/version-tracker PROJECT=kubernetes-sigs/kind VERBOSITY=6` shows `2024-05-31T18:04:24.095Z        V6      Pushing changes to remote [eks-distro-pr-bot]`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Common.mk [hunk 1]
```diff
@@ -132,7 +132,8 @@ else ifneq ($(IS_RELEASE_BRANCH_BUILD),)
 	RELEASE_TARGETS_OVERRIDE=release/release-branches/all
 
 	# avoid warnings when trying to read GIT_TAG file which wont exist when no release_branch is given
-	GIT_TAG=non-existent
+	# the GIT_TAG could be set during version-tracker's upgrade command, so no override
+	GIT_TAG?=non-existent
```
Comment (jaxesn) [inline Common.mk:136]: i dont see why this is needed since youve added this target to the above allowlist, it shouldnt even fall into here?

Comment (d8660091) [reply Common.mk:135]: I tried without this "?" mark, but the command `make upgrade -C tools/version-tracker PROJECT=kubernetes-sigs/kind VERBOSITY=6` stuck at `/usr/bin/bash -c make -C /home/ubuntu/eks-anywhere-build-tooling/tools/version-tracker/eks-anywhere-build-tooling/projects/kubernetes-sigs/kind patch-repo` command, without output. I guess it's trying to wait for the non-exist tag

Comment (d8660091) [reply Common.mk:135]: I think the GIT_TAG=non-exist overrode the GIT_TAG set here: https://github.com/aws/eks-anywhere-build-tooling/blob/main/projects/kubernetes-sigs/kind/Makefile#L2

**Review by jaxesn — APPROVED** [review]

**Pull request merged**